### PR TITLE
[MEP]Default Value

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo [xinguo.li@zilliz.com](mailto:xinguo.li@zilliz.com)
#23337

### CODE REVIEW

File: docs/design_docs/20230405-default_value.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+## Design Details
+
+1. Add the default_value in the field schema as an optional field.
+
+```proto
+message FieldSchema {
+  ...
+  repeated schema.FieldData default_value = 10;
```
Comment (congqixia) [inline docs/design_docs/20230405-default_value.md:38]: why shall `default_value` be repeated here?

File: docs/design_docs/20230405-default_value.md
Comment (congqixia) [inline docs/design_docs/20230405-default_value.md:1]: Shall we support default value for vector column?

Comment (smellthemoon) [reply docs/design_docs/20230405-default_value.md:1]: yes

File: docs/design_docs/20230405-default_value.md (hunk 1 above)
Comment (smellthemoon) [reply docs/design_docs/20230405-default_value.md:38]: Sorry, it's a mistake, I will correct it.

File: docs/design_docs/20230405-default_value.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# MEP: Default Value
+
+Current state: Under Discussion
+
+ISSUE: [[Feature]: Support Default Value #23337](https://github.com/milvus-io/milvus/issues/23337)
+
+Keywords: Default, Insert, Upsert 
+
+Released: v2.4.0
```
Comment (jiaoew1991) [inline docs/design_docs/20230405-default_value.md:9]: Bring it on 2.3.1, don't wait for 2.4.0

File: docs/design_docs/20230405-default_value.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+- Test for using default value in proxy
+
+
+### E2E Tests
+|                  Test Cases                  |         Expected Behavior          |
+| :------------------------------------------: | :--------------------------------: |
+|         set illegal default value            |            report error            |
+|          set legal default value             |  use default value as fields data  |
+|          don't set default value             |              the same              |
```
Comment (jiaoew1991) [inline docs/design_docs/20230405-default_value.md:72]: Add some compatibility tests, the schema built in 2.2 should be able to be used normally in the new version

File: docs/design_docs/20230405-default_value.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+message InsertRequest {
+  ... 
+  repeated schema.FieldData fields_data = 5;
+}
+
+message UpsertRequest {
+  ... 
+  repeated schema.FieldData fields_data = 5;
+}
+```
```
Comment (jiaoew1991) [inline docs/design_docs/20230405-default_value.md:54]: Add an example with pymilvus

File: docs/design_docs/20230405-default_value.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+```python
+    # create collection
+    nb = 3000
+    default_fields = [
+        FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True),
+        FieldSchema(name="float", dtype=DataType.FLOAT,
+                    default_value=[np.float32(i) for i in range(nb)]),
```
Comment (yah01) [inline docs/design_docs/20230405-default_value.md:62]: how could this default value be a array? what will happen if the number of rows of inserted columns is not matched with the default value's?

It should be just a value, not array

**Review by yah01 — CHANGES_REQUESTED** [review]

File: docs/design_docs/20230405-default_value.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+message UpsertRequest {
+  ... 
+  repeated schema.FieldData fields_data = 5;
+}
+```
+
+```python
+    # create collection
+    nb = 3000
+    default_fields = [
```
Comment (yah01) [inline docs/design_docs/20230405-default_value.md:59]: `default_fields` -> `fields`

**Review by yah01 — CHANGES_REQUESTED** [review]

File: docs/design_docs/20230405-default_value.md
Comment (smellthemoon) [reply docs/design_docs/20230405-default_value.md:1]: default_value only support scalars for now

**Pull request merged**